### PR TITLE
fix dimension mismatch in DownsamplerBlock 

### PR DESCRIPTION
**Pull request by IDKiro**

### CODE REVIEW

**Comment by xiaoyufenfei** [comment]
Hi ,thanks you suggestion!
I have a problem. In the downsampling unit, one branch uses an average pool of stride=2 and kernel_size=2, and the other branch uses a kernel_size=2 convolution(stride=2), due to the resolution and number of channels of the two branches all same, why do you say that the sampling unit dimensions do not match? I am looking forward to your reply.

**Comment by IDKiro** [comment]
I think it is because different kernel size of convolution and pooling lead to a different result of output resolution. 

I'm glad to provide my test code for you. Just change patch_size to 31 make the error occur.

```
from __future__ import division
from __future__ import print_function
import os, time, scipy.io, shutil
import torch
import torch.nn as nn
from torch.autograd import Variable
import torch.nn.functional as F
import numpy as np


class DownsamplerBlock (nn.Module):
    def __init__(self, in_channel, out_channel):
        super().__init__()

        self.conv = nn.Conv2d(in_channel, out_channel-in_channel, (3, 3), stride=2, padding=1, bias=True)
        self.pool = nn.MaxPool2d(2, stride=2)
        self.bn = nn.BatchNorm2d(out_channel, epatch_size=1e-3)
        self.relu = nn.ReLU(inplace=True)

    def forward(self, input):
        output = torch.cat([self.conv(input), self.pool(input)], 1)
        output = self.bn(output)
        output = self.relu(output)
        return output 


patch_size = 32                 # right when patch size is 32 but wrong when patch size is 31

inch = 16
outch = 32

test_img = np.random.rand(inch, patch_size, patch_size)

input_var = torch.autograd.Variable(
        torch.from_numpy(test_img.copy()).type(torch.FloatTensor).unsqueeze(0)
    )

model = DownsamplerBlock(inch, outch)

output = model(input_var)
```

**Comment by xiaoyufenfei** [comment]
Thank you for your reply. I don't quite understand, is there a necessity to do this? Because no errors are found at runtime, if you are not sure if the resolution is the same of the two branch, just print the size of the tensor to see if there is a problem.   Since Pytorch is very flexible, if you want to achieve downsampling by convolution, such as convolution kernel kernel_size=K, we only need to set stride=2, padding=(K-1)/2
As follows：

class DownsamplerBlock (nn.Module):
    def __init__(self, in_channel, out_channel):
        super().__init__()

        self.conv = nn.Conv2d(in_channel, out_channel-in_channel, (3, 3), stride=2, padding=1, bias=True)
        self.pool = nn.MaxPool2d(2, stride=2)
        self.bn = nn.BatchNorm2d(out_channel, epatch_size=1e-3)
        self.relu = nn.ReLU(inplace=True)

    def forward(self, input):
        ##
        print(self.conv(input).shape)
        print(self.pool(input).shape)
        ##
        output = torch.cat([self.conv(input), self.pool(input)], 1)
        output = self.bn(output)
        output = self.relu(output)
        return output

**Comment by IDKiro** [comment]
yes, it's no need to do this for task.
I just think it will be easier to apply the network module to other datasets or tasks.
Thank you for your reply.
